src/schemas/tokens-info.ts

```diff
@@ -26,34 +26,61 @@ const checkMinVersion: CustomValidator = (value) => {
   }
 }
 
-const TokensInfoSchema = Joi.object().pattern(
-  AddressSchema,
+const BaseTokenInfoSchema = Joi.object({
+  address: AddressSchema,
+  imageUrl: Joi.string()
+    // For now only allow assets within this repo
+    .pattern(
+      /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
+    )
+    .uri()
+    .custom(checkMatchingAsset, 'has a matching asset'),
+  name: Joi.string().required(),
+  decimals: Joi.number().required(),
+  symbol: Joi.string().required(),
+  isCoreToken: Joi.boolean(),
+  isSupercharged: Joi.boolean(),
+  pegTo: AddressSchema,
+  // Forcing flag to be false if decimals is not 18
+  isSwappable: Joi.when('decimals', {
+    is: 18,
+    then: Joi.boolean(),
+    otherwise: Joi.valid(false),
+  }),
+  minimumAppVersionToSwap: Joi.string()
+    .pattern(/^\d+\.\d+\.\d+$/)
+    .custom(checkMinVersion, 'has a valid version'),
+  isNative: Joi.boolean(),
+})
+
+export const TokenInfoSchema = Joi.alternatives().try(
+  Joi.object({
+    // native tokens don't have an address except CELO
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.string().invalid('CELO').required(),
+    address: Joi.forbidden(),
+  }).concat(BaseTokenInfoSchema),
   Joi.object({
+    // CELO is native and has an address
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.valid('CELO').required(),
     address: AddressSchema.required(),
-    imageUrl: Joi.string()
-      // For now only allow assets within this repo
-      .pattern(
-        /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
-      )
-      .uri()
-      .custom(checkMatchingAsset, 'has a matching asset'),
-    name: Joi.string().required(),
-    decimals: Joi.number().required(),
-    symbol: Joi.string().required(),
-    isCoreToken: Joi.boolean(),
-    isSupercharged: Joi.boolean(),
-    // This checks it's referencing a token address that exists in the root object
-    pegTo: Joi.valid(Joi.ref('/', { in: true })),
-    // Forcing flag to be false if decimals is not 18
-    isSwappable: Joi.when('decimals', {
-      is: 18,
-      then: Joi.boolean(),
-      otherwise: Joi.valid(false),
-    }),
-    minimumAppVersionToSwap: Joi.string()
-      .pattern(/^\d+\.\d+\.\d+$/)
-      .custom(checkMinVersion, 'has a valid version'),
-  }),
+  }).concat(BaseTokenInfoSchema),
+  Joi.object({
+    // all non-native tokens require address
+    isNative: Joi.boolean().invalid(true),
+    symbol: Joi.string().invalid('CELO').required(),
```

Comment:
Why the `invalid('CELO')` in needed?
Won't this disallow `CELO` tokens in other chains? I can imagine a wrapped CELO with that symbol.

Reply:
That is a good point. I don't think it's needed, since `isNative: Joi.boolean().invalid(true)` should already exclude CELO from this case. I was just trying to make sure native CELO was required to have `isNative: true`

Reply:
but you're right, we could have non-native tokens with symbol CELO, too. Similar to how we have ETH tokens that are non-native.

Reply:
that said, since CELO is just one case and shouldn't ever be updated, I think it's probly ok to forsake coverage and just manually check it. We could even add a bespoke jest test if we were worried enough (I'm not, though)